x/recovery/keeper/ibc_callbacks_integration_test.go

```diff
@@ -0,0 +1,460 @@
+
+				// Check that aevmos were not recovered
+				coin := suite.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(suite.EvmosChain.GetContext(), senderAcc, "aevmos")
+				suite.Require().Equal(coin, sdk.NewCoin("aevmos", sdk.NewInt(10000)))
+				// Check that uosmo were not deposited since sender has a claims record (with nothing claimed)
+				coins := suite.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(suite.EvmosChain.GetContext(), receiverAcc, uosmoIbcdenom)
+				suite.Require().Equal(coins.Amount, sdk.NewInt(0))
+			},
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			suite.SetupTest()
+			path := suite.pathOsmosisEvmos
+
+			tc.malleate()
+			suite.SendAndReceiveMessage(path, suite.IBCOsmosisChain, "uosmo", 10, sender, receiver, 1)
+			timeout = uint64(suite.EvmosChain.GetContext().BlockTime().Add(time.Hour * 4).Add(time.Second * -20).UnixNano())
+
+			tc.test()
+		})
+	}
+}
+
+// Send uatom from Cosmos to Evmos
+// Enable Withdraw
+// Send uosmo From Osmosis to Evmos
+// Aevmos, uosmo should be on Osmosis balance
+// ibc/uatom should remain on the EvmosChain
+// Send uosmo From Osmosis to Evmos
+// No changes on balance should occur
+func (suite *IBCTestingSuite) TestTwoChains() {
+	suite.SetupTest()
+
+	params := types.DefaultParams()
+	params.EnableRecovery = false
+	suite.EvmosChain.App.(*app.Evmos).RecoveryKeeper.SetParams(suite.EvmosChain.GetContext(), params)
+
+	sender := suite.IBCOsmosisChain.SenderAccount.GetAddress().String()
+	receiver := suite.IBCOsmosisChain.SenderAccount.GetAddress().String()
+	pathOsmosisEvmos := suite.pathOsmosisEvmos
+	pathCosmosEvmos := suite.pathCosmosEvmos
+
+	suite.SendAndReceiveMessage(pathCosmosEvmos, suite.IBCCosmosChain, "uatom", 10, suite.IBCCosmosChain.SenderAccount.GetAddress().String(), receiver, 1)
+
+	params.EnableRecovery = true
+	suite.EvmosChain.App.(*app.Evmos).RecoveryKeeper.SetParams(suite.EvmosChain.GetContext(), params)
+
+	suite.SendAndReceiveMessage(pathOsmosisEvmos, suite.IBCOsmosisChain, "uosmo", 10, sender, receiver, 1)
+	timeout := uint64(suite.EvmosChain.GetContext().BlockTime().Add(time.Hour * 4).Add(time.Second * -20).UnixNano())
+
+	// Recreate packets that were sent in the ibc_callback
+	// Coins locked
+	packet2 := CreatePacket("10000", "aevmos", sender, receiver,
+		"transfer", "channel-0", "transfer", "channel-0", 1, timeout)
+
+	packet3 := CreatePacket("10", "transfer/channel-0/uosmo", sender, receiver,
+		"transfer", "channel-0", "transfer", "channel-0", 2, timeout)
+
+	// Relay both packets that were sent in the ibc_callback
+	err := pathOsmosisEvmos.RelayPacket(packet2)
+	suite.Require().NoError(err)
+	err = pathOsmosisEvmos.RelayPacket(packet3)
+	suite.Require().NoError(err)
+
+	senderAcc, err := sdk.AccAddressFromBech32(sender)
+	suite.Require().NoError(err)
+	receiverAcc, err := sdk.AccAddressFromBech32(receiver)
+	suite.Require().NoError(err)
+
+	// Aevmos was recovered from user address
```

Comment:
It would be easier to read if put the checks for one denom together, e.g.
```
	// Aevmos was recovered from user address and is available on IBCOsmosisChain
	coin := suite.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(suite.EvmosChain.GetContext(), senderAcc, "aevmos")
	suite.Require().Equal(coin, sdk.NewCoin("aevmos", sdk.NewInt(0)))
	coin = suite.IBCOsmosisChain.GetSimApp().BankKeeper.GetBalance(suite.IBCOsmosisChain.GetContext(), receiverAcc, aevmosIbcdenom)
	suite.Require().Equal(coin.Amount, sdk.NewInt(10000))

```